src/firefly/java/edu/caltech/ipac/table/io/VoTableWriter.java

```diff
@@ -0,0 +1,443 @@
+     * @throws IOException on error
+     */
+    public static void save(File file, DataGroup dataGroup, String outputFormat, boolean forExport, boolean isGenericOutput)
+            throws IOException {
+
+        FitsFactory.useThreadLocalSettings(true);  // consistent with FitsTableReader
+        FitsFactory.setLongStringsEnabled(false);
+
+        StarTable st = formStarTableFrom(dataGroup, forExport, outputFormat);
+
+        if (isGenericOutput) {
+            writeGenericVotable(file, outputFormat, st);
+        } else {
+            writeVotable(new FileWriter(file), outputFormat, st);
+        }
+    }
+
+
+    /**
+     * save the catalogs to a stream, stream is not closed
+     *
+     * @param stream the output stream to write to
+     *
+     * @param dataGroup data group
+     * @param outputFormat votable output format
+     * @throws IOException on error
+     */
+    public static void save(OutputStream stream, DataGroup dataGroup, String outputFormat, boolean forExport, boolean isGenericOutput)
+            throws IOException {
+
+        FitsFactory.useThreadLocalSettings(true);  // consistent with FitsTableReader
+        FitsFactory.setLongStringsEnabled(false);
+
+        StarTable st = formStarTableFrom(dataGroup, forExport, outputFormat);
+
+        if (isGenericOutput) {
+            writeGenericVotable(stream, outputFormat, st);
+        } else {
+            writeVotable(new OutputStreamWriter(stream), outputFormat, st);
+        }
+    }
+
+    private static List<DataType> getColumnsForVotable(DataGroup dataGroup, boolean forExport) {
+
+        List<DataType> headers = Arrays.asList(dataGroup.getDataDefinitions());
+
+        if (forExport) {
+            // this should return only visible columns
+            headers = headers.stream()
+                    .filter(dt -> IpacTableUtil.isVisible(dataGroup, dt)
+                            && !dt.getKeyName().equals(DataGroup.ROW_IDX)
+                            && !dt.getKeyName().equals(DataGroup.ROW_NUM))
+                    .collect(Collectors.toList());
+        }
+
+        return headers;
+    }
+
+    private static StarTable formStarTableFrom(DataGroup dataGroup, boolean forExport, String outputFormat) {
+
+        List<DataType> headers = getColumnsForVotable(dataGroup, forExport);
+        List<ColumnInfo> colList = headers.stream()
+                .map((dt) -> DataGroupStarTable.convertToColumnInfo(dt, outputFormat))
+                .collect(Collectors.toList());
+
+        ColumnInfo[] colInfos = colList.toArray(new ColumnInfo[colList.size()]);
+        return new DataGroupStarTable(dataGroup, colInfos, headers);
+    }
+
+
+    private static void writeGenericVotable(File file, String outputFormat, StarTable st) throws IOException
+    {
+        VOTableWriter voWriter = new VOTableWriter(getDataFormat(outputFormat), true, getVotVersion(outputFormat));
+        voWriter.writeStarTable(st, file.getName(), new StarTableOutput());
```

Comment:
As is, no difference between this and `writeGenericVotable(OutputStream stream`.

Reply:
the parameter passed is different. (voWriter.writeStarTable(st, stream); // for stream